### PR TITLE
add create instance tests

### PR DESCRIPTION
**Pull request by piyusharorawork**
Just added few tests for create instance to be consistent with other tests . 

@kof  Let me know if there are any issues with pr

### CODE REVIEW

File: app/shared/tree-utils/create-instance.test.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { createInstance } from "./create-instance";
+
+describe("create instance", () => {
+  test("instance without id should be auto generated", () => {
+    const createdInstance = createInstance({
+      component: "Box",
+      id: undefined,
```
Comment (kof) [inline app/shared/tree-utils/create-instance.test.ts:7]: can just omit the id

File: app/shared/tree-utils/create-instance.test.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      children: [],
+      style: {},
```
Comment (kof) [inline app/shared/tree-utils/create-instance.test.ts:9]: can omit style and children

File: app/shared/tree-utils/create-instance.test.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    });
+
+    // We cannot do snapshot testing here as the id will always be unique
+    expect(createdInstance.id).toBeDefined();
```
Comment (kof) [inline app/shared/tree-utils/create-instance.test.ts:13]: to be a string

File: app/shared/tree-utils/create-instance.test.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        {
+          id: "1",
+          component: "Box",
+          style: {},
+          children: [],
+        },
+      ],
+      style: {},
+    });
+    expect(createdInstance.children.length).toBe(1);
```
Comment (kof) [inline app/shared/tree-utils/create-instance.test.ts:40]: would want to check if the exact same child was preserved for e.g.

File: app/shared/tree-utils/create-instance.test.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      id: undefined,
+      children: [
+        {
+          id: "1",
+          component: "Box",
+          style: {},
+          children: [],
+        },
+      ],
+      style: {},
```
Comment (kof) [inline app/shared/tree-utils/create-instance.test.ts:38]: would add a test in the same style that checks style is an object after creation without passing a style

**Review by kof — CHANGES_REQUESTED** [review]
looking good, left a few comments

**Comment by kof** [comment]
merged, thank you!

**Comment by piyusharorawork** [comment]
> merged, thank you!

@kof  I wanted to contribute more to this project , please let me know if you want me to work on any specific issue which you   are not already working on .

**Comment by kof** [comment]
@piyusharorawork do you have some particular topic or area of interest? Let's chat on discord, I need to find for you something that fits your interest, experience and time!

**Pull request merged**